src/dex/solidly/solidly-e2e.test.ts

```diff
@@ -0,0 +1,613 @@
+              network,
+              provider,
+            );
+          });
+        });
+
+        describe('megaSwap', () => {
+          it('MATIC -> TOKEN', async () => {
+            await testE2E(
+              tokens.USDT,
+              tokens.MATIC,
+              holders.USDT,
+              usdAmount,
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.megaSwap,
+              network,
+              provider,
+            );
+          });
+
+          it('Token -> MATIC', async () => {
+            await testE2E(
+              tokens.USDT,
+              tokens.MATIC,
+              holders.USDT,
+              usdAmount,
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.megaSwap,
+              network,
+              provider,
+            );
+          });
+
+          it('Token -> Token', async () => {
+            await testE2E(
+              tokens.WMATIC,
+              tokens.WETH,
+              holders.WMATIC,
+              maticAmount,
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.megaSwap,
+              network,
+              provider,
+            );
+          });
+        });
+      });
+
+      describe('Dystopia Stable Pools', () => {
+        describe('simpleSwap', () => {
+          it('Token -> Token', async () => {
+            await testE2E(
+              tokens.USDC,
+              tokens.USDT,
+              holders.USDC,
+              usdAmount,
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.simpleSwap,
+              network,
+              provider,
+            );
+          });
+        });
+
+        describe('multiSwap', () => {
+          it('Token -> Token', async () => {
+            await testE2E(
+              tokens.USDC,
+              tokens.USDT,
+              holders.USDC,
+              usdAmount,
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.multiSwap,
+              network,
+              provider,
+            );
+          });
+        });
+
+        describe('megaSwap', () => {
+          it('Token -> Token', async () => {
+            await testE2E(
+              tokens.USDC,
+              tokens.USDT,
+              holders.USDC,
+              usdAmount,
+              SwapSide.SELL,
+              dexKey,
+              ContractMethod.megaSwap,
+              network,
+              provider,
+            );
+          });
+        });
+      });
+    });
+  });
+
+  describe('Optimism', () => {
+    const network = Network.OPTIMISM;
+    const tokens = Tokens[network];
+    const holders = Holders[network];
+    const provider = new StaticJsonRpcProvider(
+      generateConfig(network).privateHttpProvider,
+      network,
+    );
+
+    describe('Velodrome', () => {
+      const dexKey = 'Velodrome';
+
+      describe('simpleSwap', () => {
+        it('NATIVE -> TOKEN', async () => {
```

Comment:
I think better to have specific names instead of generic TOKEN because it is easy to run particular tests